torch/csrc/jit/script/compiler.cpp

```diff
@@ -440,40 +418,16 @@ struct Environment {
     return getSugaredVar(ident, range)->asValue(range, method);
   }
 
-  // Given that after emitting statements in a block, we've added block inputs
-  // for all value references and assignments, delete inputs for which there was
-  // no assignment, only references.
-  void deleteExtraInputs(const SourceRange& loc) {
-    // note: skip i == 0, it is the loop trip count for inputs
-    // and the loop condition for outputs.
-    // captured_inputs is indexed by i - 1 since it only contains loop
-    // carried dependencies
-    //          inputs: loop_counter, lcd0, lcd1, ...
-    //         outputs: loop_condition, lcd0, lcd1, ...
-    // captured_inputs: lcd0, lcd1, ...
-    AT_ASSERT(b->inputs().size() == b->outputs().size());
-    AT_ASSERT(b->inputs().size() == captured_inputs.size() + 1);
-    for (size_t i = b->inputs().size() - 1; i > 0; i--) {
-      // nothing changed along this loop
-      if (b->inputs()[i] == b->outputs()[i]) {
-        auto name = captured_inputs[i - 1];
-        Value* orig = findInParentFrame(name)->asValue(loc, method);
-        b->inputs()[i]->replaceAllUsesWith(orig);
-        b->eraseInput(i);
-        b->eraseOutput(i);
-        captured_inputs.erase(captured_inputs.begin() + i - 1);
-      }
-    }
-  }
   std::vector<std::string> definedVariables() {
     std::vector<std::string> result;
-    for (auto& kv : value_table) {
+    for (auto& kv : first_class_types) {
       result.push_back(kv.first);
     }
     return result;
   }
 
  private:
+  SimpleValueTable first_class_types;
```

Comment:
This should be a map from name to type and it isn't for really confusing reasons.